### ISSUE TITLE
Why are search results displayed as an ordered list?

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#4089]
Search results in Backdrop are displayed as an ordered (numbered) list. Why is that?

It appears to have [originated](https://github.com/backdrop/backdrop/commit/ffb760fa4978dd812469c99006e3bfc56290470a) in 2010 in an effort to improve accessibility, citing Google and Yahoo's style. However neither Google nor Yahoo use an ordered list anymore.

Also, and IMO more importantly, since the ordered list numbering is reset on each page of results, it looks weird and pointless:

![search_results](https://user-images.githubusercontent.com/2385329/65599977-c6d6a900-dfa7-11e9-91cd-cba947217968.png)
(Note that `page=1` is actually the *2nd* page of results)
In this case I'd expect Job Scheduler to be numbered as '26' (since the first 25 results were on page 1).

So I suggest changing the `<ol>` to an `<ul>`.

**Comment by klonos** [comment]
This is the respective d.org issue where it all happened: https://www.drupal.org/project/drupal/issues/810176

...that issue is titled `Search result titles should be enclosed in a heading tag` and the intended change was to add `<h3>`'s around the links in the list. That was the accessibility improvement, and that is mentioned as what's being done by search engines:

> This seems to be the standard for search engine results (see Google, Yahoo, Bing). It allows for easy keyboard navigation with accessibility software.

Having said that, I quickly skimmed through the issue, but could not understand if this is still a thing a decade later 🤔

**Comment by klonos** [comment]
...some comments from that issue:

> I checked the spec and dt elements can only contain inline elements.
> 
> So, should we switch the search results to a different type of list?

> Google and Yahoo use an ordered list. Bing uses and unordered list. I think I prefer the Google/Yahoo structure.

...I have checked, and although Yahoo have kept the `<ol>`, Google seem to have switched to using `<div>`s

**Comment by jenlampton** [comment]
Just adding a note here that Google search results are not accessable at all, and that we should not be using what they are doing as a good example :)

**Comment by BWPanda** [comment]
> Just adding a note here that Google search results are not accessable at all

@jenlampton Do you have a source for that claim? I just find it hard to believe that the largest search engine in the world has search results that "are not accessable *at all*"...

In any case, the suggestion here is not to use divs as they do, but to use an *unordered* list.

**Comment by indigoxela** [comment]
> ... the suggestion here is not to use divs as they do, but to use an unordered list.

@BWPanda I get your argument re paginated content, but is this based on any known WGAC recommendation, or best practices, or just based on an assumption?

There's for sure a lot we can (and have to) do to improve Backdrop's out-of-the-box accessibility. But I doubt that switching from `ol` to `ul` in search results will help much.

**Comment by BWPanda** [comment]
It's just an opinion and personal preference. I don't see the point of using an ordered list for search results. If someone tells me it's better for accessibility, then fine. At least then we have a reason for it. But AFAIK, there's no reason for it...

And sure, we could just leave it as-is. Except that paginated results then look weird and broken. So IMO an unordered list is a better fit.

**Comment by izmeez** [comment]
I think an unordered list would be ok, but an ordered list needs the numbers to be sequential on subsequent pages as mentioned in https://github.com/backdrop/backdrop-issues/issues/7160
